osu.Game.Rulesets.Osu/Difficulty/Skills/Rhythm.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Rulesets.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Osu.Objects;
+
+namespace osu.Game.Rulesets.Osu.Difficulty.Skills
+{
+    public class Rhythm : OsuStrainSkill
+    {
+        private const double rhythm_multiplier = 0.75;
+        private const int history_time_max = 5000; // 5 seconds of calculatingRhythmBonus max.
+
+        private double strainDecayBase => 0.3;
+        private double currentStrain;
+
+        protected override double DecayWeight => 1.0;
+        protected override double DifficultyMultiplier => 4;
+        protected override int HistoryLength => 32;
+
+        private readonly double greatWindow;
+
+        public Rhythm(Mod[] mods, double hitWindowGreat)
+            : base(mods)
+        {
+            greatWindow = hitWindowGreat;
+        }
+
+        /// <summary>
+        /// Calculates a rhythm multiplier for the difficulty of the tap associated with historic data of the current <see cref="OsuDifficultyHitObject"/>.
+        /// </summary>
+        private double calculateRhythmBonus(DifficultyHitObject current)
+        {
+            if (current.BaseObject is Spinner)
+                return 0;
```

Comment:
Rhythm multiplier has a range of `[1, Infinity)`, so shouldn't this return 1? Otherwise, it can result in a negative strain value because the spinner would be adding -1 to `currentStrain` as per `StrainValueAt`. Even more so, the idea of negative strain is weird.
```suggestion
            if (current.BaseObject is Spinner)
                return 1;
```
An example situation is when a spinner appears after a long break, which isn't uncommon. `currentStrain` would be very small due to strain decay to the point where adding -1 will result in a negative number.